docs/docs/config/layout.md

```diff
@@ -0,0 +1,225 @@
+
+| Property   | Type     | Description                          | Unit                                                    |
+| ---------- | -------- | ------------------------------------ | ------------------------------------------------------- |
+| Width      | int (>0) | Key(cap) width                       | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Height     | int (>0) | Key(cap) height                      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| X          | uint     | Key X position (top-left point)      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Y          | uint     | Key Y position (top-left point)      | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Rotation   | int      | Key rotation (positive => clockwise) | [centi-](https://en.wikipedia.org/wiki/Centi-)degree    |
+| Rotation X | int      | Rotation origin X position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+| Rotation Y | int      | Rotation origin Y position           | [centi-](https://en.wikipedia.org/wiki/Centi-)"keyunit" |
+
+The `key_physical_attrs` node is defined in [`dts/physical_layouts.dtsi`](https://github.com/zmkfirmware/zmk/blob/main/app/dts/physical_layouts.dtsi) and is mandatory.
+
+## Physical Layout Position Map
+
+Defines a mapping between [physical layouts](#physical-layout), allowing key mappings to be preserved in the same locations as previously when using [ZMK Studio](../features/studio.md). Read through the [page on physical layouts](../development/hardware-integration/physical-layouts.md) for more information.
+
+### Devicetree
+
+Applies to: `compatible = zmk,physical-layout-position-map`
+
+Definition file: [zmk/app/dts/bindings/zmk,physical-layout-position-map.yaml](https://github.com/zmkfirmware/zmk/blob/main/app/dts/bindings/zmk%2Cphysical-layout-position-map.yaml)
+
+| Property   | Type    | Description                                                                                      | Default |
+| ---------- | ------- | ------------------------------------------------------------------------------------------------ | ------- |
+| `complete` | boolean | If the mapping complete describes the key mapping, and no position based mapping should be used. |         |
+
+The `zmk,physical-layout-position-map` node should have one child node per physical layout. Each child node should have the following properties:
+
+| Property          | Type    | Description                                                                       | Default |
+| ----------------- | ------- | --------------------------------------------------------------------------------- | ------- |
+| `physical-layout` | phandle | The physical layout that corresponds to this mapping entry                        |         |
+| `positions`       | array   | Array of key positions that match the same array entry in the other sibling nodes |         |
+
+## Matrix Transform
```

Comment:
IMHO, the matrix transform information should be first, since physical layouts are mainly relevant for ZMK Studio, which isn't fully released yet, and not all boards/shields need to support out of the gate today.

Reply:
That's fair, I'll make that change. I think in the future we will want them swapped back though.